### DISCUSSION TITLE
Target Controlled Infusion

### DISCUSSION THREAD

**Discussion by Lucas-Wamser** [Open-Systems-Pharmacology/Forum#1624] [Questions & Problems]
My name is Lucas Wamser, and I am a veterinarian working on a research team in Brazil. We recently developed a PBPK model for a specific anesthetic in dogs. We are now interested in using this model to control an infusion pump.

To do this, we would like to know if there is a way to define a target plasma concentration and if the model could calculate the infusion rates needed to maintain this concentration over time. Doing this would be the opposite of the conventional procedure since we usually set a dose, and the model calculates the resulting concentration based on it. 

Can we accomplish this using the OSPsuit package, Javascript, PK-Sim, or MObi? There may be another way.
Would anyone have the knowledge or experience to help us solve this issue? 

Thank you for any guidance or assistance you can offer.

**Comment by Yuri05** [comment]
If you have a **predefined** infusion schema (e.g. _new infusion starts every 15 minutes_ or so), you could use a [**Parameter Identification Tool**](https://docs.open-systems-pharmacology.org/shared-tools-and-example-workflows/parameter-identification) in PK-Sim or MoBi for doing this.
You can [import your target concentration as an observed data set with constant concentration](https://docs.open-systems-pharmacology.org/shared-tools-and-example-workflows/import-edit-observed-data), and then optimize doses and infusion times for each administration.

### Example parameter identification with 3 infusions every 8 hours
![grafik](https://github.com/Open-Systems-Pharmacology/Forum/assets/25061876/c623e665-5cca-4f29-92e3-31a75393ea0a)

![grafik](https://github.com/Open-Systems-Pharmacology/Forum/assets/25061876/ba2f8b90-0302-4a50-a4fd-7419ebcdc33a)

### Comparison of the concentration with the target concentration over 24 hours
![grafik](https://github.com/Open-Systems-Pharmacology/Forum/assets/25061876/cdb4d420-4d68-4144-a456-a8d24d1205e9)

If you have a **free** infusion schema (infusion pump starts "on demand"), it becomes a more challenging task.
You still could solve this task with a predefined infusion schema (where infusions are defined very frequently) - however, the parameter optimization might not deliver good results in this case.
Other solutions via events definition in MoBi, more sophisticated optimization via R etc. could be developed as well.

**Reply by Lucas-Wamser** [reply]
Of course, I understand. Thank you for your reply. The challenge with identifying the parameters is related to the time-consuming execution and the impossibility of modifying the target at any time. Is it possible to solve this by changing the differential equations in MATLAB?

**Reply by StephanSchaller** [reply]
Hi @Lucas-Wamser,

If I understand you right, you would like to have the opportunity to optimize the infusion per patient and adjust the target concentration at a given point (e.g., based on the response, you might want to reduce or increase the target).
This could be quite easily done by coupling the PBPK model with a closed-loop controller in R, which is, in principle, nothing more than a way to identify the infusion rate based on a customizable target concentration using an optimization method (basically an adaptive parameter estimation of the infusion rate parameter given a changing target).

This is called "model-predictive control" and sth. we implemented for the continuous (pump-) titration of insulin in Type 1 Diabetes: https://ieeexplore.ieee.org/document/7315018

We did it in Matlab back then, but this is now "easily" done in R + R Shiny.

Best,
Stephan
